cpp/models/abm/location.cpp

```diff
@@ -34,210 +33,144 @@ namespace abm
 Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
     : m_type(type)
     , m_index(index)
-    , m_capacity(LocationCapacity())
     , m_capacity_adapted_transmission_risk(false)
-    , m_subpopulations(Eigen::Index(InfectionState::Count))
-    , m_cached_exposure_rate({AgeGroup::Count, VaccinationState::Count})
     , m_cells(std::vector<Cell>(num_cells))
     , m_required_mask(MaskType::Community)
     , m_npi_active(false)
 {
-    m_subpopulations.add_time_point(0);
-    m_subpopulations.get_last_value().setZero();
 }
 
-InfectionState Location::interact(const Person& person, TimeSpan dt,
-                                  const GlobalInfectionParameters& global_params) const
+void Location::interact(Person& person, TimePoint t, TimeSpan dt, const GlobalInfectionParameters& global_params) const
 {
-    auto infection_state       = person.get_infection_state();
-    auto vaccination_state     = person.get_vaccination_state();
-    auto age                   = person.get_age();
-    ScalarType mask_protection = person.get_protective_factor(global_params);
-    switch (infection_state) {
-    case InfectionState::Susceptible:
-        if (!person.get_cells().empty()) {
-            for (auto cell_index : person.get_cells()) {
-                InfectionState new_state = random_transition(
-                    infection_state, dt,
-                    {{InfectionState::Exposed,
-                      (1 - mask_protection) * m_cells[cell_index].cached_exposure_rate[{age, vaccination_state}]}});
-                if (new_state != infection_state) {
-                    return new_state;
-                }
+    auto age               = person.get_age();
+    double mask_protection = person.get_mask_protective_factor(global_params);
```

Comment:
Maybe you can start switching double to ScalarType. Please check other instances.

Reply:
I changed a lot of doubles. There may still are some in the simulation and the tests, which I will change later.